synapse/app/_base.py

```diff
@@ -106,6 +106,9 @@ def register_sighup(func: Callable[P, None], *args: P.args, **kwargs: P.kwargs)
 def start_worker_reactor(
     appname: str,
     config: HomeServerConfig,
+    # We deliberately bind this later so that we don't rely on the run() method
+    # being available at import time.
```

Comment:
I don't think this is quite accurate. It's not the availablity of `run()` that is the problem, it is *which* `run` we bind to.

```suggestion
    # Use a lambda to avoid binding to a given reactor at import time.
```

Reply:
Well, it's sort of both. It binds to `None` if it's too early, so it's also not available :-)